go/common/crypto/address/version.go

```diff
@@ -0,0 +1,28 @@
+package address
+
+import (
+	"fmt"
+	"sync"
+)
+
+// Version is an address version.
+type Version uint8
+
+func (v Version) MarshalBinary() (data []byte, err error) {
+	data = []byte{(byte)(v)}
+	return
+}
+
+var registeredVersions sync.Map
+
+// NewVersion creates and registers a new version. This routine will panic if
+// the version is already registered.
+func NewVersion(rawVersion uint8) Version {
```

Comment:
Hm, so versions are global? That is slightly strange as you can imagine having different kinds of addresses and versioning should probably be within each kind.

Reply:
Agreed. I've changed that.